### ISSUE TITLE
Filtering problems in token-ring

### ISSUE THREAD

**Issue by guyharris** [the-tcpdump-group/tcpdump#4]
Converted from [SourceForge issue 574488](http://sourceforge.net/support/tracker.php?aid=574488), submitted by nobody

Trying to dump with a filer like 
tcpdump -X -i tr0 -n -s 4056 ip host xxx.xxx.xxx.xxx
I do not dump anything.
The reason seems to be the fact that the host I'm
trying to trace (and which is connecting to the host
i'm running tcpdump on ) is on a remote eth networkm
while i'm on a token ring one.

the output i get if I dump without filters is:

07:05:00.975449 rt = 610:11:ff00 (1500) stfsrv01.http &gt;
stftk06.49454: S 1959913187:1959913187(0) ack 101 win
12048 &lt;mss 4016&gt; (DF)
0x0000   aaaa 0300 0000 0800 4500 002c 0000 4000  
 ........E..,..@.
0x0010   4006 efd3 8493 a160 8494 a070 0050 c12e  
 @......`...p.P..
0x0020   74d1 e6e3 0000 0065 6012 2f10 f678 0000      
 t......e`./..x..
0x0030   0204 0fb0  
 ....
07:05:01.033334 0:3:91:0:71:48 sap 90 &gt;
0:40:aa:7d:bb:4b ip-sap I (s=0,r=8,P) len=56
0x0000   0690 0011 ff00 aaaa 0300 0000 0800 4500  
 ..............E.
0x0010   0028 00c8 0000 3a06 3510 8494 a070 8493  
 .(....:.5....p..
0x0020   a160 c12e 0050 0000 0065 74d1 e6e4 5010  
 .`...P...et...P.
0x0030   060c 4136 0000 0000 0000 0000  
 ..A6........

In other words, those frames coming from eth make
tcpdump unable to read layer 3 headers ( i may assume
because of the rt value, don't know...).
Note that the previous dump was about a
SYN-ACK -&gt; 
ACK &lt;-
between the same two machines.

Emanuele &quot;SKULL&quot; Balla
skull@skullkrusher.net

**Comment by infrastation** [comment]
This is the oldest open bug in this bug tracker (originally opened on 2002-06-27). The proposed workaround for source-routed Token Ring packets was not implemented since then, as far as I understand `print-token.c` git history. Since it does not make much difference in modern networks, it would be acceptable if it never happens. @guyharris, if you still have that .pcap file and are determined to apply the suggested change, please tell. Otherwise this bug will be closed in 7 days.
